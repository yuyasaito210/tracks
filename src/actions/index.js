import AssessmentActions from "/actions/assessmentActions"
import AuthActions from "/actions/authActions"
import CourseActions from "/actions/courseActions"
import GlobalActions from "/actions/globalActions"
import IntegrationActions from "/actions/integrationActions"
import UserActions from "/actions/userActions"

export {
  AssessmentActions,
  AuthActions,
  CourseActions,
  GlobalActions,
  IntegrationActions,
  UserActions,
}
