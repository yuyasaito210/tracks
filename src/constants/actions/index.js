import AssessmentAction from "/constants/actions/assessment"
import AuthAction from "/constants/actions/auth"
import CourseAction from "/constants/actions/course"
import ExternalAction from "/constants/actions/external"
import GlobalAction from "/constants/actions/global"
import IntegrationAction from "/constants/actions/integration"
import UserAction from "/constants/actions/user"

const AppActions = Object.assign(
  {},
  AssessmentAction,
  AuthAction,
  CourseAction,
  ExternalAction,
  GlobalAction,
  IntegrationAction,
  UserAction,
)

export default AppActions
