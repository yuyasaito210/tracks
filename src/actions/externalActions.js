import { AppActions } from "/constants"

export function updateExternalData(data) {
  return {
    type: AppActions.UPDATE_EXTERNAL,
    ...data,
  }
}
