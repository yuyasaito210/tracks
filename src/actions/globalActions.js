import { AppActions } from "/constants"
import { GlobalService } from "/services"

const globalService = new GlobalService()

export function showLoader(loadingMessage = "") {
  return {
    type: AppActions.SHOW_SPINNER,
    loadingMessage,
  }
}

export function hideLoader() {
  return {
    type: AppActions.HIDE_SPINNER,
  }
}

export function togglePageScroll(isAllowedScroll) {
  return {
    type: AppActions.TOGGLE_PAGE_SCROLL,
    isAllowedScroll,
  }
}

export function connectionError(isInternetConnected) {
  return {
    type: AppActions.CONNECTION_ERROR,
    isInternetConnected,
  }
}

export function serverError() {
  return {
    type: AppActions.SERVER_ERROR,
  }
}

export function recordMobileUserEvent(type, data = {}, courseId) {
  const body = {
    type,
    course_id: courseId,
    data,
  }

  return dispatch => globalService.createMobileUserEvent(body)
}
