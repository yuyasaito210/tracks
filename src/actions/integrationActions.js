import { AppActions } from '/constants'
import { IntegrationService } from '/services'

const integrationService = new IntegrationService()

function receiveIntegrationStatus (profile, allowNotifications) {
  return {
    type: AppActions.RECEIVE_INTEGRATION_STATUS,
    profile,
    allowNotifications
  }
}

function receiveIntegrationStatusFailed () {
  return {
    type: AppActions.RECEIVE_INTEGRATION_STATUS_FAILED
  }
}

function updateIntegrationNotificationPermissionSuccessful (allowNotifications) {
  return {
    type: AppActions.UPDATE_INTEGRATION_NOTIFICATION_PERMISSION,
    allowNotifications
  }
}

function updateIntegrationNotificationPermissionFailed () {
  return {
    type: AppActions.UPDATE_INTEGRATION_NOTIFICATION_PERMISSION_FAILED
  }
}

function deleteIntegrationSuccessful () {
  return {
    type: AppActions.DELETE_INTEGRATION
  }
}

function deleteIntegrationFailed () {
  return {
    type: AppActions.DELETE_INTEGRATION_FAILED
  }
}

export function getIntegrationStatus () {
  return (dispatch) => {
    integrationService.getIntegrationStatus()
      .then((res) => {
        if (res.status === 200) {
          return res.json()
            .then((body) => {
              dispatch(receiveIntegrationStatus(body.integration_profile, body.allow_notifications))
            })
        }
        return dispatch(receiveIntegrationStatusFailed())
      })
      .catch((e) => {
        dispatch(receiveIntegrationStatusFailed(e))
      })
  }
}

export function updateIntegrationNotificationPermission (allowNotifications) {
  const body = {
    allow_mobile_notifications: allowNotifications
  }
  return (dispatch) => {
    integrationService.updateIntegration(body)
      .then((res) => {
        if (res.status === 200) {
          dispatch(updateIntegrationNotificationPermissionSuccessful(allowNotifications))
        } else {
          updateIntegrationNotificationPermissionFailed()
        }
      })
      .catch((e) => {
        dispatch(updateIntegrationNotificationPermissionFailed(e))
      })
  }
}

export function deleteIntegration () {
  return (dispatch) => {
    integrationService.deleteIntegration()
      .then((res) => {
        if (res.status === 200) {
          dispatch(deleteIntegrationSuccessful())
        } else {
          dispatch(deleteIntegrationFailed())
        }
      })
      .catch((e) => {
        dispatch(deleteIntegrationFailed(e))
      })
  }
}
