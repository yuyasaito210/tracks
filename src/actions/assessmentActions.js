import cloneDeep from "lodash.clonedeep"
import AssessmentAttempt from "assessment-js"
import { AsyncStorage } from "react-native"
import { Actions, ActionConst } from "react-native-router-flux"
import { AppActions } from "/constants"
import { AssessmentService } from "/services"
import * as GlobalActions from "/actions/globalActions"

const assessmentService = new AssessmentService()

function completingAttemptComplete(body) {
  return {
    type: AppActions.ASSESSMENT_COMPLETING_ATTEMPT_COMPLETE,
    payload: body,
  }
}

function completingAttemptStarting() {
  return {
    type: AppActions.ASSESSMENT_COMPLETING_ATTEMPT_STARTING,
  }
}

function getAssessmentIntroSuccessful(body, item) {
  return {
    type: AppActions.ASSESSMENT_INTRO,
    assessmentIntro: body,
    assessmentInfo: item,
  }
}
function getAssessmentResultSuccessful(body) {
  return {
    type: AppActions.ASSESSMENT_RESULT_DATA,
    payload: {
      assessmentResults: body,
    },
  }
}
function getAssessmentIntroFailed(error) {
  return {
    type: AppActions.ASSESSMENT_INTRO_FAILED,
    payload: {
      error,
    },
  }
}

function getDynamicDataSuccessful(body) {
  return {
    type: AppActions.ASSESSMENT_DYNAMIC_DATA,
    dynamicData: body,
  }
}

function getDynamicDataFailed(error) {
  return {
    type: AppActions.ASSESSMENT_DYNAMIC_DATA_FAILED,
    payload: {
      error,
    },
  }
}

function navigationSuccessful(assessment) {
  return {
    type: AppActions.ASSESSMENT_NAVIGATION_SUCCESSFUL,
    payload: assessment.jsonify(),
  }
}

function newAssessmentAttemptSuccessful(assessmentBody) {
  const assessment = new AssessmentAttempt(assessmentBody)
  assessment.fastForward()
  AsyncStorage.setItem("pageLoadTime", `${Date.now()}`)
  return {
    type: AppActions.NEW_ASSESSMENT_ATTEMPT,
    newAssessmentAttempt: assessment,
  }
}

function newAssessmentAttemptFailed(error) {
  return {
    type: AppActions.NEW_ASSESSMENT_ATTEMPT_FAILED,
    payload: {
      error,
    },
  }
}

function waitOnDynamicData() {
  return {
    type: AppActions.WAIT_ON_DYNAMIC_DATA,
  }
}

function disableResultsPage() {
  return {
    type: AppActions.DISABLE_RESULT_PAGE,
  }
}

function checkAnswersSuccessful(assessment) {
  return {
    type: AppActions.ASSESSMENT_CHECK_ANSWERS,
    payload: assessment.responses,
  }
}

function submitAnswersComplete() {
  return {
    type: AppActions.ASSESSMENT_SUBMIT_ANSWERS_COMPLETE,
  }
}

function submitAnswersStarting() {
  return {
    type: AppActions.ASSESSMENT_SUBMIT_ANSWERS_STARTING,
  }
}

function visitedPage(visitedPages) {
  return {
    type: AppActions.ASSESSMENT_VISIT_PAGE,
    visitedPages,
  }
}

async function updateAnalytics(assessment) {
  const analytics = {
    analytics: {
      page_id: assessment.currentPage().id,
      load_time: await AsyncStorage.getItem("pageLoadTime"),
      submit_time: Date.now(),
    },
  }
  assessmentService.putAnalytics(assessment.id, analytics)
  const newLoadTime = Date.now()
  await AsyncStorage.setItem("pageLoadTime", `${newLoadTime}`)
  assessment.updateAnalytics(analytics.analytics)
  return assessment
}

export function visitPage(assessment, pageId) {
  return (dispatch) => {
    assessmentService.visitPage(assessment.id, { page_id: pageId })
    const newAssessment = cloneDeep(assessment)
    newAssessment.visitPage(pageId)
    dispatch(visitedPage(Array.from(newAssessment.visitedPages)))
  }
}

export function nextPage(assessment) {
  return async (dispatch) => {
    const newAssessment = await updateAnalytics(cloneDeep(assessment))
    newAssessment.navigateToNextPage()
    dispatch(navigationSuccessful(newAssessment))
  }
}

export function previousPage(assessment) {
  return async (dispatch) => {
    const newAssessment = await updateAnalytics(cloneDeep(assessment))
    newAssessment.navigateToPreviousPage()
    dispatch(navigationSuccessful(newAssessment))
  }
}

export function navigateToSectionId(assessment, sectionId) {
  return async (dispatch) => {
    const newAssessment = await updateAnalytics(cloneDeep(assessment))
    newAssessment.navigateToSectionId(sectionId)
    dispatch(navigationSuccessful(newAssessment))
  }
}

export function toggleNavButtons(isEnabled) {
  return {
    type: AppActions.TOGGLE_NAV_BUTTONS,
    isEnabled,
  }
}
export function setCurrentPage(page) {
  return {
    type: AppActions.SET_CURRENT_PAGE,
    page,
  }
}
export function saveResponse(questionId, value, feedbackText) {
  return {
    type: AppActions.SAVE_ANSWER,
    payload: {
      questionId,
      value,
      feedbackText,
    },
  }
}
export function onSlider(title, type, value) {
  return {
    type: AppActions.CHANGE_PRIORITY_DATA,
    payload: {
      title,
      type,
      value,
    },
  }
}

export function deleteResponse(questionId) {
  return {
    type: AppActions.DELETE_ANSWER,
    payload: {
      questionId,
    },
  }
}

export function setStandardResponse(questionId, value) {
  return {
    type: AppActions.SET_STANDARD_RESPONSE,
    questionId,
    value,
  }
}

export function checkAnswers(assessment, responses) {
  return (dispatch) => {
    const newAssessment = cloneDeep(assessment)
    const body = {
      instant_feedback: Object.keys(responses),
      question: responses,
    }

    assessmentService.checkAnswers(assessment.id, body)
                     .then((res) => {
                       dispatch(submitAnswersComplete())
                       if (res.status !== 204) {
                         dispatch(GlobalActions.serverError())
                         Actions.ErrorView()
                         return null
                       }
                       return true
                     })

    Object.keys(responses).forEach(questionId => newAssessment.writeResponse(questionId, responses[questionId]))
    dispatch(checkAnswersSuccessful(newAssessment))
  }
}

export function submitResponses(assessment, responses, goToNextPage = false) {
  return (dispatch) => {
    dispatch(submitAnswersStarting())
    const newAssessment = cloneDeep(assessment)
    const body = { question: responses }
    assessmentService.putAnswer(assessment.id, body)
                     .then((res) => {
                       dispatch(submitAnswersComplete())
                       if (res.status !== 204) {
                         dispatch(GlobalActions.serverError())
                         Actions.ErrorView()
                         return null
                       }
                       return true
                     })
    Object.keys(responses).forEach(questionId => newAssessment.writeResponse(questionId, responses[questionId]))
    dispatch(checkAnswersSuccessful(newAssessment))
    if (goToNextPage) {
      dispatch(nextPage(newAssessment))
    }
  }
}

export function sendResponce(assessmentAttemptId, questionId, value, feedbackText) {
  const body = {
    question: {
      [questionId]: value.toString(),
    },
  }
  return (dispatch) => {
    dispatch(saveResponse(questionId, value, feedbackText))
    assessmentService.putAnswer(assessmentAttemptId, body)
                     .then((res) => {
                       if (res.status !== 204) {
                         dispatch(getAssessmentIntroFailed())
                         dispatch(deleteResponse(questionId))
                         dispatch(GlobalActions.serverError())
                         Actions.ErrorView()
                       }
                     })
                     .catch((e) => {
                       dispatch(getAssessmentIntroFailed(e))
                       dispatch(deleteResponse(questionId))
                       dispatch(GlobalActions.serverError())
                       Actions.ErrorView()
                     })
  }
}

export function getDynamicData(attemptId, key) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader())
    dispatch(waitOnDynamicData())
    assessmentService.getDynamicData(attemptId, key)
                     .then(res => res.json()
                                     .then((body) => {
                                       dispatch(GlobalActions.hideLoader())
                                       dispatch(getDynamicDataSuccessful(body))
                                     }))
                     .catch((e) => {
                       dispatch(GlobalActions.hideLoader())
                       dispatch(getDynamicDataFailed(e))
                       dispatch(GlobalActions.serverError())
                       Actions.ErrorView()
                     })
  }
}

export function getCalculatedResults(attemptId) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader())
    dispatch(disableResultsPage())
    assessmentService.getResults(attemptId)
                     .then(res => res.json()
                                     .then((body) => {
                                       dispatch(GlobalActions.hideLoader())
                                       dispatch(getAssessmentResultSuccessful(body))
                                     }))
                     .catch((e) => {
                       dispatch(GlobalActions.hideLoader())
                       dispatch(getAssessmentResultSuccessful({}))
                       dispatch(getAssessmentIntroFailed(e))
                       dispatch(GlobalActions.serverError())
                       Actions.ErrorView()
                     })
  }
}

export function renderAssessmentIntro(item, navigateViaPop = false) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader())
    return assessmentService
      .getAssessmentIntro(item.url)
      .then(res =>
        res.json()
           .then((body) => {
             dispatch(getAssessmentIntroSuccessful(body, item))
             dispatch(GlobalActions.hideLoader())
             if (navigateViaPop) {
               return Actions.pop()
             }
             return Actions.AssessmentItem({
               title: body.introduction.introPage.title,
               type: ActionConst.PUSH,
             })
           }))
      .catch((e) => {
        dispatch(getAssessmentIntroFailed(e))
        dispatch(GlobalActions.hideLoader())
        dispatch(GlobalActions.serverError())
        Actions.ErrorView()
      })
  }
}

export function startNewAssessment(assessmentIntro, courseId, sectionId, assignmentId, assignmentName) {
  const requestBody = {
    assessment_id: assessmentIntro.assessment_id,
  }

  if (courseId !== undefined && courseId !== null && courseId !== "") {
    requestBody.course_id = courseId
  }
  if (assignmentId !== undefined && assignmentId !== null && assignmentId !== "") {
    requestBody.assignment_id = assignmentId
  }
  if (sectionId !== undefined && sectionId !== null && sectionId !== "") {
    requestBody.section_id = sectionId
  }
  return (dispatch) => {
    dispatch(GlobalActions.showLoader())
    dispatch(GlobalActions.recordMobileUserEvent("AssignmentStarted", { id: assignmentId, name: assignmentName }, courseId))
    assessmentService.newAssessmentAttempt(requestBody)
                     .then(res => res.json()
                                     .then((body) => {
                                       dispatch(newAssessmentAttemptSuccessful(body))
                                       dispatch(GlobalActions.hideLoader())
                                       Actions.NewAssessment({
                                         title: assessmentIntro.introduction.introPage.title,
                                         type: ActionConst.PUSH,
                                       })
                                     }))
                     .catch((e) => {
                       dispatch(newAssessmentAttemptFailed(e))
                       dispatch(GlobalActions.hideLoader())
                       dispatch(GlobalActions.serverError())
                       Actions.ErrorView()
                     })
  }
}

export function completeAssessment(assessment) {
  return (dispatch) => {
    dispatch(completingAttemptStarting())
    assessmentService.completeAttempt(assessment.id)
                     .then(res =>
                       res.json()
                          .then((body) => {
                            dispatch(completingAttemptComplete(body))
                          }))
  }
}

export function continueAttempt(url, assessmentId, title, analyticsData = {}) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader())
    if (assessmentId !== 1) {
      dispatch(GlobalActions.recordMobileUserEvent("AssignmentContinued", { id: analyticsData.id, name: analyticsData.name }, analyticsData.courseId))
      return assessmentService.continueAttempt(url)
        .then(res => res.json()
            .then((body) => {
              dispatch(newAssessmentAttemptSuccessful(body))
              dispatch(GlobalActions.hideLoader())
              return Actions.NewAssessment({ title, type: ActionConst.PUSH })
            }))
        .catch((e) => {
          dispatch(newAssessmentAttemptFailed(e))
          dispatch(GlobalActions.hideLoader())
          dispatch(GlobalActions.serverError())
          Actions.ErrorView()
        })
    }
    dispatch(newAssessmentAttemptSuccessful("WBA is under construction"))
    dispatch(GlobalActions.hideLoader())
    return Actions.NewAssessment({ title: "WBA", type: ActionConst.RESET })
  }
}

export function hideGlossary() {
  Actions.refresh({ backTitle: "History" })

  return dispatch => dispatch({
    type: AppActions.ASSESSMENT_HIDE_GLOSSARY,
  })
}

export function showGlossary() {
  Actions.refresh({ backTitle: "Chapter", title: "Glossary" })

  return dispatch => dispatch({
    type: AppActions.ASSESSMENT_SHOW_GLOSSARY,
  })
}

export function hideTOC() {
  Actions.refresh({ backTitle: "History" })

  return dispatch => dispatch({
    type: AppActions.ASSESSMENT_HIDE_TOC,
  })
}

export function showTOC() {
  Actions.refresh({ backTitle: "Chapter", title: "TOC" })

  return dispatch => dispatch({
    type: AppActions.ASSESSMENT_SHOW_TOC,
  })
}
