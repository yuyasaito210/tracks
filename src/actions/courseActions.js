import { Actions, ActionConst } from "react-native-router-flux"
import { AppActions } from "/constants"
import { CourseService } from "/services"
import * as GlobalActions from "/actions/globalActions"

const courseService = new CourseService()

function receiveAnnouncementsBegin() {
  return {
    type: AppActions.RECEIVE_ANNOUNCEMENTS_BEGIN,
  }
}

function receiveAnnouncementsFailed() {
  return {
    type: AppActions.RECEIVE_ANNOUNCEMENTS_FAILED,
  }
}

function receiveAnnouncementsSuccessful(announcements) {
  return {
    type: AppActions.RECEIVE_ANNOUNCEMENTS_SUCCESSFUL,
    announcements,
  }
}

function receiveAssignmentBegin() {
  return {
    type: AppActions.RECEIVE_ASSIGNMENT_BEGIN,
  }
}

function receiveAssignmentFailed() {
  return {
    type: AppActions.RECEIVE_ASSIGNMENT_FAILED,
  }
}

function receiveAssignmentSuccessful(assignment) {
  return {
    type: AppActions.RECEIVE_ASSIGNMENT_SUCCESSFUL,
    assignment,
  }
}

function receiveCourseBegin() {
  return {
    type: AppActions.RECEIVE_COURSE_BEGIN,
  }
}

function receiveCourseFailed() {
  return {
    type: AppActions.RECEIVE_COURSE_FAILED,
  }
}

function receiveCourseSuccessful(course) {
  return {
    type: AppActions.RECEIVE_COURSE_SUCCESSFUL,
    course,
  }
}

function receiveCoursesBegin() {
  return {
    type: AppActions.RECEIVE_COURSES_BEGIN,
  }
}

function receiveCoursesSuccessful(courses) {
  return {
    type: AppActions.RECEIVE_COURSES_SUCCESSFUL,
    courses,
  }
}

export function selectAnnouncement(announcement) {
  Actions.Announcement({ type: ActionConst.RESET })
  return {
    type: AppActions.SELECT_ANNOUNCEMENT,
    announcement,
  }
}

export function selectAssignment(assignment, navigationOptions = {}) {
  Actions.Assignment(navigationOptions)
  return {
    type: AppActions.SELECT_ASSIGNMENT,
    assignment,
    backTitle: navigationOptions.backTitle,
  }
}

export function selectCourse(course) {
  return {
    type: AppActions.SELECT_COURSE,
    course,
  }
}

export function selectCourseById(courseId) {
  return {
    type: AppActions.SELECT_COURSE_BY_ID,
    courseId,
  }
}

function deleteAnnouncementSuccessful(announcementId) {
  return {
    type: AppActions.DELETE_ANNOUNCEMENT,
    announcementId,
  }
}

function markAnnouncementReadFailed(announcementId) {
  return {
    type: AppActions.MARK_ANNOUNCEMENT_READ_STATUS,
    announcementId,
    read: false,
  }
}

function markAnnouncementReadSuccessful(announcementId) {
  return {
    type: AppActions.MARK_ANNOUNCEMENT_READ_STATUS,
    announcementId,
    read: true,
  }
}

function markAnnouncementUnreadFailed(announcementId) {
  return markAnnouncementReadSuccessful(announcementId)
}

function markAnnouncementUnreadSuccessful(announcementId) {
  return markAnnouncementReadFailed(announcementId)
}

export function goToCourse(course) {
  return (dispatch) => {
    dispatch(selectCourse(course))
    return Actions.Tabbar({ type: ActionConst.RESET })
  }
}

export function getCourse(courseId) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader("Loading course data"))
    dispatch(receiveCourseBegin())
    return courseService
      .getCourse(courseId)
      .then((res) => {
        dispatch(GlobalActions.hideLoader())
        if (res.status === 200) {
          return res.json()
                    .then(body => (
                      dispatch(receiveCourseSuccessful(body))
                    ))
        }
        return dispatch(receiveCourseFailed())
      })
  }
}

export function getCourses() {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader("Loading list of courses"))
    dispatch(receiveCoursesBegin())
    return courseService
      .getCourses()
      .then((res) => {
        dispatch(GlobalActions.hideLoader())
        if (res.status === 200) {
          return res.json()
                    .then((body) => {
                      dispatch(receiveCoursesSuccessful(body))
                      if (body.length === 1) {
                        return goToCourse(body[0])
                      }
                      return Actions.Courses({
                        type: ActionConst.RESET,
                      })
                    })
        }
        dispatch(GlobalActions.serverError())
        return Actions.ErrorView()
      })
  }
}

export function deleteAnnouncement(announcementId, navigate = false) {
  return (dispatch) => {
    dispatch(deleteAnnouncementSuccessful(announcementId))
    if (navigate) {
      Actions.Announcements({ type: ActionConst.RESET })
    }
    return courseService.deleteAnnouncement(announcementId)
  }
}

export function getAnnouncements(courseId) {
  return (dispatch) => {
    dispatch(receiveAnnouncementsBegin())
    return courseService
      .getAnnouncements(courseId)
      .then((res) => {
        if (res.status === 200) {
          return res.json()
                    .then(body => (
                      dispatch(receiveAnnouncementsSuccessful(body))
                    ))
        }
        return dispatch(receiveAnnouncementsFailed())
      })
  }
}

export function markAnnouncementRead(courseId, announcementId) {
  const body = {
    course_id: courseId,
    read: true,
  }
  return (dispatch) => {
    dispatch(markAnnouncementReadSuccessful(announcementId))
    return courseService
      .markAnnouncementReadStatus(announcementId, body)
      .then((res) => {
        if (res.status !== 204) {
          return dispatch(markAnnouncementReadFailed(announcementId))
        }
        return true
      })
  }
}

export function markAnnouncementUnread(courseId, announcementId) {
  const body = {
    course_id: courseId,
    read: false,
  }
  return (dispatch) => {
    dispatch(markAnnouncementUnreadSuccessful(announcementId))
    return courseService
      .markAnnouncementReadStatus(announcementId, body)
      .then((res) => {
        if (res.status !== 204) {
          return dispatch(markAnnouncementUnreadFailed(announcementId))
        }
        return true
      })
  }
}

export function getAssignment(assignmentId, courseId, navigate = false) {
  return (dispatch) => {
    dispatch(GlobalActions.showLoader("Refreshing assignment data"))
    dispatch(receiveAssignmentBegin())
    if (navigate) {
      Actions.pop()
    }
    return courseService
      .getAssignment(assignmentId, courseId)
      .then((res) => {
        dispatch(GlobalActions.hideLoader())
        if (res.status === 200) {
          return res.json()
                    .then(body => (
                      dispatch(receiveAssignmentSuccessful(body))
                    ))
        }
        return dispatch(receiveAssignmentFailed())
      })
  }
}
