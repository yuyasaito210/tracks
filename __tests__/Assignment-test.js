import "react-native"
import React from "react"
import renderer from "react-test-renderer"
import mockStore from "redux-mock-store"
import { selectAssignment } from "../src/actions/courseActions"
import AssignmentView from "../src/components/Assignment"

describe("should handle", () => {
  const store = mockStore({})

  beforeEach(() => {
    store.clearActions()
  })

  const assignment = {
    id: 1,
    course_id: 1,
    name: "Sample Assignment 1",
    description: "",
    due_date: "2031-07-20T19:59:00Z",
    points_possible: 5,
    available_on_mobile: true,
    position: 1,
    url: "https://beartracks-test.perceivant.com/api/v1/assessments/sample_assessment_1",
    grade: "0/5",
    late: false,
    missing: false,
    past_due: false,
    status: "",
    submitted: false,
    to_do: true,
  }

  describe("index component", () => {
    const renderAssessmentIntro = jest.fn()
    it("renders as expected", () => {
      const component = renderer.create(
        <AssignmentView
          assignment={assignment}
          renderAssessmentIntro={renderAssessmentIntro} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()

      component.getInstance().goToAssignment(assignment)
      expect(renderAssessmentIntro).toHaveBeenCalledWith(assignment)
      expect(renderAssessmentIntro).toHaveBeenCalledTimes(1)
    })
  })
})
