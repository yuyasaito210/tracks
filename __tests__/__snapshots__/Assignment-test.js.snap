// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle index component renders as expected 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "height": 1232,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EE6A3C",
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "space-around",
          "paddingHorizontal": 30,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 45,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "BearTracksIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 15,
            }
          }
        >
          This assignment is due:
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": 10,
            }
          }
        >
          Sunday, July 20th
        </Text>
      </View>
    </View>
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#2F3C47",
              "fontWeight": "bold",
              "paddingBottom": 10,
              "paddingTop": 30,
              "textAlign": "center",
            }
          }
        >
          Sample Assignment 1
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#2F3C47",
              "fontWeight": "bold",
              "paddingBottom": 10,
              "textAlign": "center",
            }
          }
        >
          Points: 5
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 0,
            "borderWidth": 0,
            "flexDirection": "column",
            "height": 160,
            "justifyContent": "center",
            "margin": 20,
            "padding": 15,
          },
          Object {
            "backgroundColor": "#FFD05B",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#2F3C47",
                  "fontSize": 45,
                },
                undefined,
                Object {
                  "fontFamily": "BearTracksIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#2F3C47",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginTop": 20,
            }
          }
        >
          Take Assessment
        </Text>
        <Text
          style={
            Object {
              "color": "#2F3C47",
              "fontSize": 15,
              "fontWeight": "300",
              "marginTop": 5,
            }
          }
        >
          Sample Assignment 1
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 25,
              },
              Object {
                "color": "#979B9C",
              },
              Object {
                "fontFamily": "BearTracksIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#979B9C",
              "paddingTop": 10,
              "textAlign": "center",
            }
          }
        >
          Assignment 
          Due Soon
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
